lib/redis.rb

```diff
@@ -2130,13 +2130,14 @@ def unwatch
 
   def pipelined
     synchronize do |client|
+      pipeline = Pipeline.new
       begin
-        original, @client = @client, Pipeline.new
+        original, @client = @client, pipeline
         yield(self)
-        original.call_pipeline(@client)
       ensure
         @client = original
       end
+      original.call_pipeline(pipeline)
```

Comment:
By reseting `@client` before `call_pipeline`, so that if another command is issued inside `call_pipeline` it isn't appended to the pipeline being executed.